src/System.Runtime.Extensions/src/System/IO/StringWriter.cs

```diff
@@ -221,6 +281,30 @@ public override Task WriteLineAsync(string value)
             return Task.CompletedTask;
         }
 
+        public override Task WriteLineAsync(StringBuilder value, CancellationToken cancellationToken = default)
+        {
+            if (GetType() != typeof(StringWriter))
+            {
+                // This overload was added after the WriteLineAsync(StringBuilder, ...) overload, and so in case
+                // a derived type may have overridden it, we need to delegate to it, which the base does.
+                return base.WriteLineAsync(value, cancellationToken);                
+            }
+
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            if (!_isOpen)
+            {
+                throw new ObjectDisposedException(null, SR.ObjectDisposed_WriterClosed);
+            }
+
+            _sb.Append(value);
+            WriteLine();
```

Comment:
@stephentoub here i copied https://github.com/MarcoRossignoli/corefx/blob/2e806cc11b226b403b772db90149c28029b80724/src/System.Runtime.Extensions/src/System/IO/StringWriter.cs#L196 but why not `_sb.AppendLine()` ?

Reply:
>Even if it existed, though, you wouldn't want to use it here, as TextWriter let's you change what the NewLine characters to be used are, and StringBuilder's AppendLine wouldn't respect that.

Understood!